MatrixKit/Controllers/MXKRoomViewController.m

```diff
@@ -918,6 +921,68 @@ - (void)setRoomInputToolbarViewClass:(Class)roomInputToolbarViewClass
     [_roomInputToolbarContainer setNeedsUpdateConstraints];
 }
 
+
+- (void)setRoomExtraInfoViewClass:(Class)roomExtraInfoViewClass
+{
+    // Sanity check: accept only MXKRoomInputToolbarView classes or sub-classes
+    NSParameterAssert([roomExtraInfoViewClass isSubclassOfClass:MXKRoomExtraInfoView.class]);
+    
+    if (!_roomExtraInfoContainer)
+    {
+        NSLog(@"[MXKRoomVC] Set setRoomExtraInfoViewClass failed: container is missing");
+        return;
+    }
+
+    // Remove potential toolbar
+    if (extraInfoView)
+    {
+        if ([NSLayoutConstraint respondsToSelector:@selector(deactivateConstraints:)])
+        {
+            [NSLayoutConstraint deactivateConstraints:extraInfoView.constraints];
+        }
+        else
+        {
+            [_roomExtraInfoContainer removeConstraints:extraInfoView.constraints];
+        }
+        [extraInfoView removeFromSuperview];
+        [extraInfoView destroy];
+    }
+    
+    extraInfoView = [roomExtraInfoViewClass roomExtraInfoView];
+    
+    // Add the view and define edge constraints
+    [_roomExtraInfoContainer addSubview:extraInfoView];
+    [_roomExtraInfoContainer addConstraint:[NSLayoutConstraint constraintWithItem:_roomExtraInfoContainer
```

Comment:
Warning: 'addConstraint' will be deprecated in a future release and should be avoided. We need to use activateConstraints instead (see deactivateConstraints vs removeConstraints just above).